### PR TITLE
Refactor resource op sample for sample test coverage

### PR DESCRIPTION
**Pull request by numerology**
Currently resource op sample illustrates the usage of mounting credential through resource op, which is not a good practice because password might not be masked in the KFP UI.

This PR changes that to an example showing how to run a container by a resource op.

Also sample test coverage is added.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2274)

### CODE REVIEW

File: samples/core/resource_ops/resource_ops.py [hunk 1]
```diff
@@ -13,45 +13,82 @@
...
+    "spec": {
+        "template": {
+            "metadata": {
+                "name": "resource-basic"
+            },
+            "spec": {
+                "containers": [{
+                    "name": "sample-container",
+                    "image": "k8s.gcr.io/busybox",
+                    "command": ["/bin/sh", "-c", "env"],
```
Comment (Ark-kun) [inline samples/core/resource_ops/resource_ops.py]: Command can just run `env` directly: `["/usr/bin/env"]`

File: samples/core/resource_ops/resource_ops.py [hunk 2]
```diff
@@ -13,45 +13,82 @@
...
+                    "envFrom": [{
+                        "configMapRef": {
+                            "name": "sample-env-config"
```
Comment (Ark-kun) [inline samples/core/resource_ops/resource_ops.py]: The name should probably come from `env_config_map_op.output`.

**Comment by Ark-kun** [comment]
`"Error from server (Forbidden): error when creating \"/tmp/manifest.yaml\": configmaps is forbidden: User \"system:serviceaccount:kubeflow:pipeline-runner\" cannot create resource \"configmaps\" in API group \"\" in the namespace \"default\"\ngithub.com/argoproj/argo/errors.New\n\t/go/src/github.com/argoproj/argo/errors/errors.go:49\ngithub.com/argoproj/argo/workflow/executor.`

File: samples/core/resource_ops/resource_ops.py [hunk 3]
```diff
@@ -13,45 +13,55 @@
...
-        k8s_resource=secret_resource,
-        attribute_outputs={"name": "{.metadata.name}"}
-    )
+def resourceop_basic():
 
-    secret = k8s_client.V1Volume(
-        name="my-secret",
-        secret=k8s_client.V1SecretVolumeSource(secret_name=rop.output)
+    # Start a container. Print out env vars.
+    cop = dsl.ResourceOp(
```
Comment (elikatsis) [inline samples/core/resource_ops/resource_ops.py]: nit: Rename from `cop` to `rop` or just `op`

Comment (numerology) [reply samples/core/resource_ops/resource_ops.py]: Done.

File: samples/core/resource_ops/resource_ops.py (hunk 2 above)
Comment (numerology) [reply samples/core/resource_ops/resource_ops.py]: Talked offline, removed configmap part.

**Pull request merged**